packages/compiler/src/lib/examples.ts

```diff
@@ -0,0 +1,173 @@
+import { Temporal } from "temporal-polyfill";
+import type { Program } from "../core/program.js";
+import type { Model, ObjectValue, Scalar, ScalarValue, Type, Value } from "../core/types.js";
+import { getEncode, type EncodeData } from "./decorators.js";
+
+/**
+ * Serialize the given TypeSpec value as a JSON object using the given type and its encoding annotations.
+ * The Value MUST be assignable to the given type.
+ */
+export function serializeValueAsJson(
```

Comment:
Should this be generalized as any serializing to values for any content type perhaps? Little bit concerned about a lot of JSON-specific logic here.

Reply:
talked offline, this is good for now and we can see to design  a more generic system where you can register different serializer